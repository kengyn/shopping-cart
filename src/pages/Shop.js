import React from "react";
import { NavLink, Outlet } from "react-router-dom";
import { items } from "../components/Items";
import "../styles/Shop.css";

const Shop = () => {
  const renderItems = () => {
    return items.map((item) => (
      <li key={item.id} className={item.id}>
        <NavLink to={`product/${item.id}`}>
          <img src={item.image} alt={item.name} />
        </NavLink>
        <div>{item.name}</div>
        <div>${item.price}</div>
      </li>
    ));
  };

  return (
    <div className="shop-main">
      <h1>Shop</h1>
      <div className="items-container">
        <ul>{renderItems()}</ul>
      </div>
      <Outlet />
    </div>
  );
};

export default Shop;
