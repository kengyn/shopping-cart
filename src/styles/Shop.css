.shop-main {
  width: 100vw;
  height: 95vh;
  position: relative;
  overflow: hidden;
  padding: 0 10vw;
}

.shop-main h1 {
  font-size: 1.3rem;
  margin-top: 1rem;
}

.items-container ul {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  width: 100%;
  gap: 1rem;
}

.items-container ul li img,
.items-container ul li {
  cursor: pointer;
  width: 250px;
  height: auto;
  transition: transform 0.2s;
}

.items-container ul li img:hover {
  transform: scale(1.05);
}

.shop-item-name {
  font-size: 1.2rem;
  margin-bottom: 0.5rem;
}
