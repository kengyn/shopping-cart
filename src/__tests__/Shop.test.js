import React from "react";
import { render } from "@testing-library/react";
import Shop from "../pages/Shop";
import { BrowserRouter as Router } from "react-router-dom";

describe("shop component", () => {
  it("renders shop items", () => {
    const { container } = render(
      <Router>
        <Shop />
      </Router>
    );

    expect(container).toMatchSnapshot();
  });
});
