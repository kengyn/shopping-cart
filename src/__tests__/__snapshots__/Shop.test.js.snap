// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shop component renders shop items 1`] = `
<div>
  <div
    class="shop-main"
  >
    <h1>
      Shop
    </h1>
    <div
      class="items-container"
    >
      <ul>
        <li
          class="acquaint-fate"
        >
          <a
            class=""
            href="/product/acquaint-fate"
          >
            <img
              alt="Acquaint Fate"
              src="Item_Acquaint_Fate.webp"
            />
            <div
              class="shop-item-name"
            >
              Acquaint Fate
            </div>
            <div
              class="shop-item-price"
            >
              $
              1
            </div>
          </a>
        </li>
        <li
          class="intertwined-fate"
        >
          <a
            class=""
            href="/product/intertwined-fate"
          >
            <img
              alt="Intertwined Fate"
              src="Item_Intertwined_Fate.webp"
            />
            <div
              class="shop-item-name"
            >
              Intertwined Fate
            </div>
            <div
              class="shop-item-price"
            >
              $
              1
            </div>
          </a>
        </li>
        <li
          class="saint-quartz"
        >
          <a
            class=""
            href="/product/saint-quartz"
          >
            <img
              alt="Saint Quartz"
              src="Saintquartz.webp"
            />
            <div
              class="shop-item-name"
            >
              Saint Quartz
            </div>
            <div
              class="shop-item-price"
            >
              $
              1
            </div>
          </a>
        </li>
        <li
          class="gem"
        >
          <a
            class=""
            href="/product/gem"
          >
            <img
              alt="Gem"
              src="daimonds.png"
            />
            <div
              class="shop-item-name"
            >
              Gem
            </div>
            <div
              class="shop-item-price"
            >
              $
              1
            </div>
          </a>
        </li>
        <li
          class="covenant-bookmark"
        >
          <a
            class=""
            href="/product/covenant-bookmark"
          >
            <img
              alt="Covenant Bookmark"
              src="Covenant-Bookmark.png"
            />
            <div
              class="shop-item-name"
            >
              Covenant Bookmark
            </div>
            <div
              class="shop-item-price"
            >
              $
              1
            </div>
          </a>
        </li>
        <li
          class="galaxy-bookmark"
        >
          <a
            class=""
            href="/product/galaxy-bookmark"
          >
            <img
              alt="Galaxy Bookmark"
              src="galaxy-bookmark.png"
            />
            <div
              class="shop-item-name"
            >
              Galaxy Bookmark
            </div>
            <div
              class="shop-item-price"
            >
              $
              1
            </div>
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
